File: 75-sort-colors/sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        // Dutch Flag algorithm
        int low,mid,high;
        low=mid=0;
        high=nums.size()-1;

        while(mid<=high){
            if (nums[mid]==1){
                mid++;
            }
            else if (nums[mid]==0){
                swap(nums[mid],nums[low]);
                low++;mid++;
            }
            else{
                swap(nums[mid],nums[high]);
                // mid++;
                high--;
            }
        }

    }
};

File: 75-sort-colors/sort-colors.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var sortColors = function(nums) {
    let low =0,mid=0,hi=nums.length-1;

    while(mid<=hi){
        if(nums[mid]==0){
           [ nums[mid],nums[low]]=[nums[low],nums[mid]];
           mid++;
           low++;
        }
        else if (nums[mid]==1){
            mid++;
        }
        else{
             [ nums[mid],nums[hi]]=[nums[hi],nums[mid]];
             hi--;
        }
    }
};